### PR TITLE
display Doctest code lens before comment

### LINKED ISSUE #4785: Doctest code lens is now mixed with function definition and attributes
The doc test code lens used to be at the top of the documentation block, but is now mixed in between the documentation, attributes, and function definition. Can this be changed back or made configurable?

Old display (expected):
image

New display (unexpected):
image

### PR DESCRIPTION
**Pull request by bnjjj**
close #4785

### CODE REVIEW

**Comment by matklad** [comment]
CI fails

**Comment by bnjjj** [comment]
@matklad Done :)

File: crates/ra_ide/src/display/navigation_target.rs [hunk 1]
```diff
@@ -150,6 +151,26 @@ impl NavigationTarget {
...
+        let name =
+            named.value.name().map(|it| it.text().clone()).unwrap_or_else(|| SmolStr::new("_"));
+        let focus_range = node.value.doc_comments().next().map(|it| it.syntax().text_range());
+        let frange = original_range(db, node.map(|it| it.syntax()));
+
+        NavigationTarget::from_syntax(
+            frange.file_id,
+            name,
+            focus_range,
+            frange.range,
```
Comment (matklad) [inline crates/ra_ide/src/display/navigation_target.rs:169]: Why does this set both `full_range` and `focus_range`? What is the semantics of focus range for a piece of doc comments? Seems like we should leave `focus_range`  as `None`

Comment (bnjjj) [reply crates/ra_ide/src/display/navigation_target.rs:167]: fixed

**Pull request merged**